engine/crates/engine-v2/schema/src/names.rs

```diff
@@ -0,0 +1,29 @@
+use crate::{FieldId, InterfaceId, ObjectId, Schema, UnionId};
+
+/// Small abstraction over the actual names to make easier to deal with
+/// renames later.
+/// Argument names aren't supported given current schema design.
+pub trait Names {
+    fn field(&self, field_id: FieldId) -> &str;
+    fn object(&self, object_id: ObjectId) -> &str;
+    fn union(&self, union_id: UnionId) -> &str;
+    fn interface(&self, interface_id: InterfaceId) -> &str;
+}
+
+impl Names for &Schema {
```

Comment:
Does this need to be implemented for `&Schema`?

```suggestion
impl Names for Schema {
```

Reply:
ah good catch, misused `dyn`.